Packs/Tripwire/Integrations/Tripwire/Tripwire.py

```diff
@@ -0,0 +1,510 @@
+        """
+        :param nodes_filter: (str) contains the filter of the request.
+        :return: the result of the http request from the api
+        """
+        return self._http_request(method='GET', url_suffix=URL_SUFFIX['NODES'].format(nodes_filter))
+
+    def get_rules(self, rules_filter: str):
+        """
+        :param rules_filter: (str) contains the filter of the request.
+        :return: the result of the http request from the api
+        """
+        return self._http_request(method='GET', url_suffix=URL_SUFFIX['RULES'].format(rules_filter))
+
+    def get_elements(self, elements_filter: str):
+        """
+        :param elements_filter: (str) contains the filter of the request.
+        :return: the result of the http request from the api
+        """
+        return self._http_request(method='GET', url_suffix=URL_SUFFIX['ELEMENTS'].format(elements_filter))
+
+    def get_versions(self, versions_filter: str):
+        """
+        :param versions_filter: (str) contains the filter of the request.
+        :return: the result of the http request from the api
+        """
+        demisto.info(f'fileters for version {versions_filter}')
+        return self._http_request(method='GET', url_suffix=URL_SUFFIX['VERSIONS'].format(versions_filter))
+
+
+''' HELPER FUNCTIONS '''
+
+
+def filter_versions(args: dict) -> str:
+    """
+    :param args: (dict) args of the command .
+    :return: (str) combination of all filter for the command.
+    """
+    filters = ""
+    if args.get('rule_oids'):
+        for rule_id in argToList(args.get('rule_oids')):
+            filters += f"ruleId={rule_id}&"
+    if args.get('rule_names'):
+        for rule_name in argToList(args.get('rule_names')):
+            filters += f"ruleName={rule_name}&"
+    if args.get('node_oids'):
+        for node_id in argToList(args.get('node_oids')):
+            filters += f"nodeId={node_id}&"
+    if args.get('version_oids'):
+        for id in argToList(args.get('version_oids')):
+            filters += f"id={id}&"
+    if args.get('element_oids'):
+        for element_id in argToList(args.get('element_oids')):
+            filters += f"elementId={element_id}&"
+    if args.get('element_names'):
+        for element_name in argToList(args.get('element_names')):
+            filters += f"elementName={element_name}&"
+    if args.get('node_names'):
+        for node_label in argToList(args.get('node_names')):
+            filters += f"nodeLabel={node_label}&"
+    if args.get('version_hashes'):
+        for hash in argToList(args.get('version_hashes')):
+            filters += f"hash={hash}&"
+    if args.get('baseline_version_ids'):
+        for baseline_version in argToList(args.get('baseline_version_ids')):
+            filters += f"baselineVersion={baseline_version}&"
+    if args.get('time_detected_range'):
+        start, end = argToList(args.get('time_detected_range'))
+        if not re.search(DATE_REG, start) and not re.search(DATE_REG, end):
```

Comment:
why did you add that? do not we check that in `parse_date_range`?

Reply:
no , `parse_date_rang` expects a relative time only  and fails on regular date time such as 2020-11-29T21:00:00Z